custom_components/frigate/media_source.py

```diff
@@ -47,6 +47,11 @@
 
 async def async_get_media_source(hass: HomeAssistant) -> MediaSource:
     """Set up Frigate media source."""
+
+    for config_entry in hass.config_entries.async_entries(DOMAIN):
```

Comment:
[With apologies in advance for re-thinking this] I think I have a much better idea. Leave this method the way it used to be, and check the option in `FrigateMediaSource`.

Suggestion on how to do this (pardon the verbosity as this file is quite complex). You'll need to add a new method such as `_isAllowedAsMediaSource` that takes a `frigate_instance_id` and returns `True` if for that instance_id, in the config entry (which can you get by calling `get_config_entry_for_frigate_instance_id`) the option is set correctly.

Then call that verification method in these places (in all cases you can get the `frigate_instance_id` from the identifer as `identifier.frigate_instance_id`).

* `async_resolve_media` [[here](https://github.com/blakeblackshear/frigate-hass-integration/blob/master/custom_components/frigate/media_source.py#L587)]
*  `async_browse_media` [[here](https://github.com/blakeblackshear/frigate-hass-integration/blob/master/custom_components/frigate/media_source.py#L617)]
*  `async_browse_media` [[here](https://github.com/blakeblackshear/frigate-hass-integration/blob/master/custom_components/frigate/media_source.py#L674)]

This is much better because if I have 3 Frigate instances, and I disable the media for 1 of them, media will still work as usual for the other 2 rather than 'all of nothing'.

Reply:
No problem, that's a much better solution, will jump on that tomorrow. 

The test will probably need to be tweaked as well